newIDE/app/src/ObjectEditor/ObjectEditorDialog.js

```diff
@@ -188,6 +196,36 @@ const InnerDialog = (props: InnerDialogProps) => {
           onUpdateBehaviorsSharedData={props.onUpdateBehaviorsSharedData}
         />
       )}
+      {currentTab === 'variables' && (
+        <Column expand noMargin>
+        <VariablesList
+        variablesContainer={props.object.getVariables()}
+        emptyExplanationMessage={
+          <Trans>
+            When you add variables to an object, any instance of the object put on the 
+            scene or created during the game will have these variables attached to it.
+          </Trans>
+        }
+        emptyExplanationSecondMessage={
+          <Trans>
+            For example, you can have a variable called 
+            Life representing the health of the object.
+          </Trans>
+        }
+        onSizeUpdated={
+          forceUpdate /*Force update to ensure dialog is properly positioned*/
+        }
+        onComputeAllVariableNames={() =>
+          EventsRootVariablesFinder.findAllObjectVariables(
+            props.project.getCurrentPlatform(),
+            props.project,
+            props.layout,
+            props.object
+        )}
```

Comment:
props.layout will never exist/be set, so this will never work properly. I think we can keep this call to findAllObjectVariables in SceneEditor/index.js and pass it as a prop:

```suggestion
        onComputeAllVariableNames={props.onComputeAllVariableNames}
```

and in SceneEditor/index.js:

```
<ObjectEditorDialog
  ...
  onComputeAllVariableNames={() => EventsRootVariablesFinder.findAllObjectVariables(...)}
/>
```

This way we avoid polluting ObjectEditorDialog by passing it the layout - which would be only useful for this and so would be meaning passing "too much information" to the component while it should not have access to the layout.